open-craft/frontend-app-discussions
# [WIP] [BB-2856] [BB-2747] [BB-2748] Topics, Posts, and Threads UI

Testing instructions:

1. Go to your LMS, enroll in a course (any course) and create several posts under one of the course section topics. Make sure one contains replies, and there is at least 1 discussion and 1 question post.
2. Copy the course key for the course you made comments in
3. Launch the frontend-app-discussions application with `npm install && npm start`
4. Navigate to `locahost:2002/discussions/COURSE_KEY/topics`. This is the Topics UI
5. Click on the topic group (with the gray headers) containing the topic you commented in
6. Click on the topic you commented in. This is the Threads UI
7. Click on your thread. This is the Posts/Comments/Replies UI

Please note that some reorganization has occurred:
* Posts, Comments and Replies all belong to Threads
  * There is 1 Post per Thread, and it is the very first user post in the Thread. The Post title is the Thread title
  * There are 0 or more Comments per Thread. These are the top level responses by users within the thread
  * There are 0 or more Replies per Comment. These are responses directly attributed to the parent comment
* Threads belong to Topics
  * There are 0 or more Threads per Topic
* Topics are the same as they have been in the past
